library/std/src/sys/unix/process/process_fuchsia.rs

```diff
@@ -233,8 +234,11 @@ impl Process {
 pub struct ExitStatus(i64);
 
 impl ExitStatus {
-    pub fn success(&self) -> bool {
-        self.code() == Some(0)
+    pub fn exit_ok(&self) -> Result<(), ExitStatusError> {
```

Comment:
One alternative I think we should consider is implementing `Try` for `ExitStatus` rather than having an explicit conversion method into `Result`. The try v2 RFC explicitly uses [return codes as one of it's examples](https://github.com/scottmcm/rfcs/blob/do-or-do-not/text/0000-try-trait-v2.md#implementing-try-for-a-non-generic-type), so this seems like a natural application of the `Try` trait to me. cc @scottmcm

Reply:
> One alternative I think we should consider is implementing `Try` for `ExitStatus` rather than having an explicit conversion method into `Result`. The try v2 RFC explicitly uses [return codes as one of it's examples](https://github.com/scottmcm/rfcs/blob/do-or-do-not/text/0000-try-trait-v2.md#implementing-try-for-a-non-generic-type), so this seems like a natural application of the `Try` trait to me. cc @scottmcm

Hmm.  I'm not sure exactly how this would work.  Presumably `command.status()?` which would have type `()` ?  This feels rather magical to me.

Also I'm not sure it's a good idea to block this on stabilisation of an in-development feature.  Currently **almost all of the examples in the documentation for `Command` are seriously broken**.  Almost none of them properly check the exit status of the subprocess.  The longer we leave this the more people are unknowingly writing badly broken programs, simply by following our documentation.

I'm told that providing a more ergonomic way to handle errors is a blocker for fixing that (#81452).  If instead a more ergonomic way of checking for these errors is going to be blocked behind stabilising `Try`, maybe we should revisit #81452 - i.e., add all the currently-needed boilerplate to all the examples.

Reply:
> Also I'm not sure it's a good idea to block this on stabilisation of an in-development feature. Currently almost all of the examples in the documentation for Command are seriously broken. Almost none of them properly check the exit status of the subprocess. The longer we leave this the more people are unknowingly writing badly broken programs, simply by following our documentation.
>
> I'm told that providing a more ergonomic way to handle errors is a blocker for fixing that (#81452). If instead a more ergonomic way of checking for these errors is going to be blocked behind stabilising Try, maybe we should revisit #81452 - i.e., add all the currently-needed boilerplate to all the examples.

That's reasonable. And I don't think stabilizing an explicit conversion method would prevent us from later adding the appropriate `Try` impl, so I'm not against adding this first, though I still feel that we should plan on adding the `Try` impl once the new trait is available on nightly.